lib/Doctrine/ORM/Tools/SchemaValidator.php

```diff
@@ -406,6 +407,25 @@ function (array $fieldMapping) use ($class): ?string {
                             );
                         }
 
+                        if (
+                            isset($fieldMapping['enumType'])
+                            && $propertyType !== $fieldMapping['enumType']
+                            && interface_exists($propertyType)
+                            && is_a($fieldMapping['enumType'], $propertyType, true)
+                        ) {
+                            if ($metadataFieldType === (string) (new ReflectionEnum($fieldMapping['enumType']))->getBackingType()) {
+                                return null;
+                            }
+
+                            return sprintf(
+                                "The field '%s#%s' has the metadata enumType '%s' that differs from the metadata field type '%s'.",
```

Comment:
So let's mention it here please?

```suggestion
                                "The field '%s#%s' has the metadata enumType '%s' with a backing type of %s that differs from the metadata field type '%s'.",
```